File: packages/java/parser-jvm-plugin-backbone/src/test/java/com/vaadin/fusion/parser/plugins/backbone/generics/GenericsMethodsEndpoint.java
package com.vaadin.fusion.parser.plugins.backbone.generics;

import java.util.List;

@Endpoint
public class GenericsMethodsEndpoint {
    public <T extends List<String>> T getList(T list) {
        return list;
    }

    public <T extends GenericsRefEntity<String>> T getRef(T ref) {
        return ref;
    }

    public <T> T getValueWithGenericType(T something) {
        return something;
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/backbone/generics/snapshots/GenericsMethodsEndpoint.ts
import { EndpointRequestInit } from "@vaadin/hilla-frontend";
import type GenericsBareRefEntity from "./com/vaadin/hilla/parser/plugins/backbone/generics/GenericsBareRefEntity.js";
import client from "./connect-client.default.js";
async function getList(list: Array<string | undefined> | undefined, init?: EndpointRequestInit): Promise<Array<string | undefined> | undefined> { return client.call("GenericsMethodsEndpoint", "getList", { list }, init); }
async function getRef(ref: GenericsBareRefEntity<string | undefined> | undefined, init?: EndpointRequestInit): Promise<GenericsBareRefEntity<string | undefined> | undefined> { return client.call("GenericsMethodsEndpoint", "getRef", { ref }, init); }
async function getValueWithGenericType(something: unknown, init?: EndpointRequestInit): Promise<unknown> { return client.call("GenericsMethodsEndpoint", "getValueWithGenericType", { something }, init); }
export { getList, getRef, getValueWithGenericType };
